### DISCUSSION TITLE
Any Method for Computing the Local Similarity Aware Loss of Two Meshes?

### DISCUSSION THREAD

**Discussion by PeizhiYan** [facebookresearch/pytorch3d#955] [General]
**Can we compute the local structure similarity of two meshes and use it as a loss function?**

For example (see the picture), we have three meshes A, B, and C. They have the same topology and exact vertex correspondence.

If we compute the mean squared error between each pair of vertices from two meshes, MSE(A, C) might be less than MSE(A, B) since A and C have similar scales.

However, from the view of local structure, A and B are more similar. 

Can I compute the co-tangent Laplacian (https://pytorch3d.readthedocs.io/en/latest/_modules/pytorch3d/ops/laplacian_matrices.html#cot_laplacian) of the meshes, and compare the MSE between the Laplacian matrices? 

![IMG_1684E28B431C-1](https://user-images.githubusercontent.com/34731246/143397981-75aaab28-a72e-4ac4-989d-c76b8ce212da.jpeg)
